### PR TITLE
New cubes nowcasting

### PR DESCRIPTION
**Pull request by cgsandford**
Changed attribute inheritance to reflect agreed strategy.  Cube creation utility was not the best tool for this job.

Testing:
 - [x] Ran tests and they passed OK

### CODE REVIEW

File: improver/nowcasting/pysteps_advection.py [hunk 1]
```diff
@@ -36,6 +36,7 @@
 from iris.coords import AuxCoord
 
 from improver.metadata.amend import (amend_attributes, set_history_attribute)
+from improver.metadata.utilities import generate_mandatory_attributes
```
Comment (gavinevans) [inline improver/nowcasting/pysteps_advection.py]: Should the call to `redirect_stdout` be removed (on line 49), given that the `redirect_stdout` function has been removed?

File: improver/nowcasting/pysteps_advection.py [hunk 2]
```diff
@@ -129,7 +130,7 @@ def _calculate_displacement(cube, interval, gridlength):
         displacement = np.array([udisp, vdisp])
         return displacement
 
-    def _reformat_analysis_cube(self):
+    def _reformat_analysis_cube(self, attribute_changes):
```
Comment (gavinevans) [inline improver/nowcasting/pysteps_advection.py]: The docstring for this method should probably be updated with an `Args` section for the `attribute_changes` argument. What do you think?

File: improver/nowcasting/pysteps_advection.py (hunk 1 above)
Comment (cgsandford) [reply improver/nowcasting/pysteps_advection.py]: The function hasn't actually been removed, it looks like it was accidentally duplicated when the change to get rid of `lib/improver` was merged in.  I just removed the old duplicate.

File: improver/nowcasting/pysteps_advection.py (hunk 2 above)
Comment (cgsandford) [reply improver/nowcasting/pysteps_advection.py]: I don't think it's necessary or particularly desirable to add tons of detail to private class method docstrings.  It's described in the `process` method (the public interface) what this argument does, and I didn't see any need to duplicate that here.

**Pull request merged**